datalake/src/main/java/com/sequenceiq/datalake/flow/detach/SdxDetachActions.java

```diff
@@ -244,43 +251,45 @@ protected void doExecute(SdxContext context, SdxEvent payload,
             @Override
             protected Object getFailurePayload(SdxEvent payload, Optional<SdxContext> flowContext,
                     Exception e) {
-                LOGGER.error("Attach new cluster state Detach action failed.", e);
-                return SdxDetachFailedEvent.from(payload, e);
+                LOGGER.error("Failed to attach new cluster during detach.", e);
+                return new SdxStartDetachRecoveryEvent(payload.getResourceId(), payload.getUserId(), e);
             }
         };
     }
 
-    @Bean(name = "SDX_ATTACH_NEW_CLUSTER_FAILED_STATE")
-    public Action<?, ?> sdxAttachNewClusterFailedAction() {
-        return new AbstractSdxAction<>(SdxDetachFailedEvent.class) {
+    @Bean(name = "SDX_DETACH_RECOVERY_STATE")
+    public Action<?, ?> sdxDetachRecoveryAction() {
+        return new AbstractSdxAction<>(SdxStartDetachRecoveryEvent.class) {
```

Comment:
is this feature available in prod? changing these stuff could cause a problem if a flow is running when we are rolling out a new version

Reply:
This feature is under entitlement and is nor available for customers yet.